hazelcast/src/main/java/com/hazelcast/internal/server/Server.java

```diff
@@ -134,4 +135,8 @@ default int connectionCount() {
      */
     void shutdown();
 
+    default boolean doAddressesMatch(Address expected, Address checked) {
+        ServerConnectionManager conManager = getConnectionManager(EndpointQualifier.MEMBER);
```

Comment:
minor: could you also add a comment here about that the conManager cannot be null when endpoint qualifier is `EndpointQualifier.MEMBER`?
AFAIU, we just add a precautionary null check here but we didn't expect this value to be null.